daemons/controld/controld_election.c

```diff
@@ -213,7 +213,9 @@ do_dc_takeover(long long action,
                enum crmd_fsa_input current_input, fsa_data_t * msg_data)
 {
     xmlNode *cib = NULL;
-    const char *cluster_type = name_for_cluster_type(get_cluster_type());
+    const enum pcmk_cluster_layer stack =
+        (enum pcmk_cluster_layer) get_cluster_type();
+    const char *stack_s = pcmk_cluster_layer_text(stack);
```

Comment:
might as well stick to the cluster_layer terminology

Reply:
yeah was just trying to keep variable names short, "layer" without "cluster_layer" feels odd

will use "cluster_layer", no big deal

Reply:
on a similar note, `pcmk_cluster_TYPE()` (emphasis mine) feels a little bit odd considering that it returns a cluster LAYER. However, I prefer that over calling the function `pcmk_cluster_layer()` which is confusing with `enum pcmk_cluster_layer`.

Could make it `pcmk_get_cluster_layer()` if we want

Reply:
I think `pcmk_get_cluster_layer` is clear enough. `pcmk_cluster_*` is helpful, but the library prefix is just `pcmk`

Reply:
yeah pcmk often misuses stack to mean the cluster layer. stack is actually cluster layer + pacemaker + resource agents + managed services

Reply:
> I think `pcmk_get_cluster_layer` is clear enough. `pcmk_cluster_*` is helpful, but the library prefix is just `pcmk`

reluctantly agree. I like the clean consistent feeling of having little sub-namespaces like this and *using them consistently*... however, it gets clunky fast with "cluster layer" and "cluster node" in particular.

The "namespace" so to speak is "cluster", and "cluster layer" and "cluster node" are really atomic terms where both words need to stay together.